### PR TITLE
Distance page

### PR DESCRIPTION
**Pull request by OlaShabalina**
## Overview of the Pull Request:
Add the Distance page with updated logic which includes:
* Changed content;
* Special characters like 'e', 'E', '+', '-' are restricted;
* The Save button is disabled unless User inputs a number;
* 'Google Maps' word - on click is redirecting user to https://www.google.com/maps ;
* Once user clicks Next - he/she will be directed to Suggestions page;
* Once user clicks Back- he/she will be directed to Confirm page;

## link to Trello card or Github issue
[Trello card - add 'Distance' page with updated logic and char limit](https://trello.com/c/r6JNNVuX)

## How Has This Been Tested?
- Manual testing;

## Pull Request Checklist:

Make sure the following checkboxes`[ ]` are checked with `x` before sending your PR -- thank you!

- [x] Have you included a link Trello card / Github issue and written sufficient description to help others review your work?
- [x] Is your pull request up-to-date with `main` branch?
- [x] Have you checked That code is well formatted? Did `npx prettier --check .` return no warnings/errors?
- [x] Have you written instructions on how to test that your changes work as expected?
- [x] Have you tested your work thoroughly on your local machine to ensure you are not breaking anything?

### CODE REVIEW

**Comment by Amutha37** [comment]
Nice work Olga.

**Comment by Amutha37** [comment]
Can we direct the google map link to "https://www.google.com/maps/dir///@" page, where it prompt to user to input the location and destination.

**Comment by OlaShabalina** [comment]
> Can we redirect the google map link to "https://www.google.com/maps/dir///@" page, where it prompt to user to input the location and destination. The link url can be hardcoded on the component href.

Good idea @Amutha37 , I didn't know that! 😄👍

**Review by hqtan — APPROVED** [review]
Nice work @OlaShabalina !

excellent

**Review by Amutha37 — APPROVED** [review]
Nice work Olga.

**Comment by hqtan** [comment]
Nice work @OlaShabalina !

I wasn't able to test using the preview deployment, that the previously entered (and saved) distance value would appear in the text box when a user revisits `form/Distance` page (probably because the Next button links to a non-existent page); but when I tested it locally (by updating the Next button link to an existing page), the distance value (previously entered by user) does appear in text box on `form/Distance` page :+1:

File: pages/form/DIstance.jsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+export default function Distance() {
+  const { answers, setAnswers } = useForm();
+  const [km, setKm] = useState(answers.km);
+
+  const saveAnswers = () => setAnswers((prev) => ({ ...prev, km: km }));
+
+  const router = useRouter();
+
+  // function to block input of special characters
+  const blockInvalidChar = e => ['e', 'E', '+', '-'].includes(e.key) && e.preventDefault();
```
[the remark this answers is not in the document]
Comment (OlaShabalina) [reply pages/form/DIstance.jsx]: Hi @Amutha37, It's a number-only field so those are the only restrictions.

**Pull request merged**